Fixed #2373: don't propagate ReadTimeoutExceptions

ekhov opened the pull request:
On idle timeout for NettyConnectionPool handle ReadTimeoutExceptions- just log them with debug level, so they don't propagate as unhandled errors further

#2373 
/claim #2373

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/zio/zio-http?pullRequest=2619) <br/>All committers have signed the CLA.

vigoo commented on project/Dependencies.scala:
```diff
@@ -16,6 +16,8 @@ object Dependencies {
 
   val scalafmt = "org.scalameta" %% "scalafmt-dynamic" % "3.7.17"
 
+  val `scala-logging` = "com.typesafe.scala-logging" %% "scala-logging"   % "3.9.5"
```
We should not depend on this library, ZIO has it's own built-in logging API

ekhov replied:
Used built-in logging

ekhov committed:
Fixed zio#2373: don't propagate ReadTimeoutExceptions

On idle timeout for NettyConnectionPool handle ReadTimeoutExceptions-
just log them with debug level, so they don't propagate as unhandled errors further

ekhov commented on zio-http/src/main/scala/zio/http/netty/client/NettyConnectionPool.scala:
```diff
@@ -127,9 +125,24 @@ object NettyConnectionPool {
     } yield result
   }
 
+  private final class ReadTimeoutErrorHandler(nettyRuntime: NettyRuntime)(implicit trace: Trace)
+      extends ChannelInboundHandlerAdapter {
```
Also I had an idea of implementing `ReadTimeoutHandler` itself and rewrite `readTimedOut` method, so instead of raising an exception, it would just write a log message. But I'm not sure which way is more correct/idiomatic

ekhov committed:
Merge branch 'main' into handle-client-idle-timeout
